src/aiori-IME.c

```diff
@@ -0,0 +1,277 @@
+/* -*- mode: c; c-basic-offset: 8; indent-tabs-mode: nil; -*-
+ * vim:expandtab:shiftwidth=8:tabstop=8:
+ */
+/******************************************************************************\
+*                                                                              *
+*        Copyright (c) 2003, The Regents of the University of California       *
+*      See the file COPYRIGHT for a complete copyright notice and license.     *
+*                                                                              *
+********************************************************************************
+*
+* Implement abstract I/O interface for DDN Infinite Memory Engine (IME).
+*
+\******************************************************************************/
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <errno.h>                                  /* sys_errlist */
+#include <fcntl.h>                                  /* IO operations */
+
+#include "ior.h"
+#include "iordef.h"
+#include "aiori.h"
+#include "utilities.h"
+#include "ime_native.h"
+
+#ifndef   O_BINARY                                  /* Required on Windows */
+#  define O_BINARY 0
+#endif
+
+/**************************** P R O T O T Y P E S *****************************/
+
+static void        *IME_Create(char *, IOR_param_t *);
+static void        *IME_Open(char *, IOR_param_t *);
+static void         IME_Close(void *, IOR_param_t *);
+static void         IME_Delete(char *, IOR_param_t *);
+static void         IME_SetVersion(IOR_param_t *);
+static void         IME_Fsync(void *, IOR_param_t *);
+static int          IME_Access(const char *, int, IOR_param_t *);
+static IOR_offset_t IME_GetFileSize(IOR_param_t *, MPI_Comm, char *);
+static IOR_offset_t IME_Xfer(int, void *, IOR_size_t *,
+                             IOR_offset_t, IOR_param_t *);
+
+/************************** D E C L A R A T I O N S ***************************/
+
+extern int      rank;
+extern int      rankOffset;
+extern int      verbose;
+extern MPI_Comm testComm;
+
+ior_aiori_t ime_aiori = {
+        .name = "IME",
+        .create = IME_Create,
+        .open = IME_Open,
+        .xfer = IME_Xfer,
+        .close = IME_Close,
+        .delete = IME_Delete,
+        .set_version = IME_SetVersion,
+        .fsync = IME_Fsync,
+        .get_file_size = IME_GetFileSize,
+        .access = IME_Access,
+};
+
+/***************************** F U N C T I O N S ******************************/
+
+int IOR_Initialize_IM(int debug_level)
```

Comment:
I'll remove `IOR_Initialize_IM` and `IOR_Finalize_IM` functions.

Reply:
Fine with me :+1: